sdk/core/core/inc/az_span_builder.h

```diff
@@ -71,6 +71,9 @@ AZ_NODISCARD az_result az_span_builder_append(az_span_builder * const self, az_s
  */
 AZ_NODISCARD az_result az_span_builder_append_byte(az_span_builder * const self, uint8_t const c);
 
+AZ_NODISCARD az_result
+az_span_builder_append_zeros(az_span_builder * const self, size_t const size);
```

Comment:
missing comments about what it does and how it is used.  Will it append a \0 * size times?   or what's the size for?  if yes, probably `count` is better than `size` ?